drivers/sensor/adltc2990/adltc2990.c

```diff
@@ -0,0 +1,485 @@
+			uint32_t v2_r2 =
+				cfg->pins_v1_v2_configuration.voltage_divider_resistors.v2_r1_r2[1];
+			voltage_divider_ratio = DIV_ROUND_CLOSEST(v2_r1 + v2_r2, v2_r2);
+			data->pins_v1_v2_values[1] =
+				adltc2990_get_property_value(dev, VOLTAGE_SINGLEENDED, V2) *
+				voltage_divider_ratio;
+		}
+
+		if (mode_v3_v4 == VOLTAGE_DIFFERENTIAL) {
+			data->pins_v3_v4_values[0] =
+				adltc2990_get_property_value(dev, VOLTAGE_DIFFERENTIAL, V3);
+		} else if (mode_v3_v4 == VOLTAGE_SINGLEENDED) {
+			uint32_t v3_r1 =
+				cfg->pins_v3_v4_configuration.voltage_divider_resistors.v3_r1_r2[0];
+			uint32_t v3_r2 =
+				cfg->pins_v3_v4_configuration.voltage_divider_resistors.v3_r1_r2[1];
+			voltage_divider_ratio = DIV_ROUND_CLOSEST(v3_r1 + v3_r2, v3_r2);
+			data->pins_v3_v4_values[0] =
+				adltc2990_get_property_value(dev, VOLTAGE_SINGLEENDED, V3) *
+				voltage_divider_ratio;
+
+			uint32_t v4_r1 =
+				cfg->pins_v3_v4_configuration.voltage_divider_resistors.v4_r1_r2[0];
+			uint32_t v4_r2 =
+				cfg->pins_v3_v4_configuration.voltage_divider_resistors.v4_r1_r2[1];
+			voltage_divider_ratio = DIV_ROUND_CLOSEST((v4_r1 + v4_r2), v4_r2);
+			data->pins_v3_v4_values[1] =
+				adltc2990_get_property_value(dev, VOLTAGE_SINGLEENDED, V4) *
+				voltage_divider_ratio;
+		}
+		break;
+	}
+	case SENSOR_CHAN_AMBIENT_TEMP: {
+		if (!(mode_v1_v2 == TEMPERATURE || mode_v3_v4 == TEMPERATURE)) {
+			LOG_ERR("Sensor is not configured to measure Ambient Temperature");
+			return -EINVAL;
+		}
+		if (mode_v1_v2 == TEMPERATURE) {
+			data->pins_v1_v2_values[0] =
+				adltc2990_get_property_value(dev, TEMPERATURE, V1);
+		}
+		if (mode_v3_v4 == TEMPERATURE) {
+			data->pins_v3_v4_values[1] =
+				adltc2990_get_property_value(dev, TEMPERATURE, V3);
+		}
+		break;
+	}
+	case SENSOR_CHAN_ALL: {
+		if (adltc2990_is_busy(dev)) {
+			LOG_INF("ADLTC2990 conversion ongoing");
+			return -EBUSY;
+		}
+		adltc2990_trigger_measurement(dev);
+		break;
+	}
+	default: {
+		LOG_ERR("does not measure channel: %d", chan);
+		return -ENOTSUP;
+	}
+	}
+
+	return 0;
+}
+
+static int adltc2990_channel_get(const struct device *dev, enum sensor_channel chan,
+				 struct sensor_value *val)
+{
+	if (val == NULL) {
+		LOG_ERR("Argument of type sensor_value* cannot be null ");
+		return -EINVAL;
+	}
+	struct adltc2990_data *data = dev->data;
+	const struct adltc2990_config *cfg = dev->config;
+	enum adltc2990_monitoring_type mode_v1_v2 = adltc2990_get_v1_v2_measurement_modes(
+		cfg->measurement_mode[1], cfg->measurement_mode[0]);
+	enum adltc2990_monitoring_type mode_v3_v4 = adltc2990_get_v3_v4_measurement_modes(
+		cfg->measurement_mode[1], cfg->measurement_mode[0]);
+
+	uint8_t offset_index = 0, num_values_v1_v2 = 0, num_values_v3_v4 = 0;
+
+	switch (chan) {
+	case SENSOR_CHAN_DIE_TEMP: {
+		val->val1 = data->internal_temperature / 10000;
+		val->val2 = data->internal_temperature % 10000;
+		LOG_INF("Internal Temperature Value is:%d.%d", val->val1, val->val2);
```

Comment:
This may get noisy too, use `LOG_DBG` for all the successful cases in this function